### PR TITLE
feat: Add message truncation to runtime observations

### PR DESCRIPTION
**Pull request by xingyaoww**
- Add MAX_CHARS_MESSAGES constant to base Runtime class
- Implement truncate_content for all runtime observations
- Apply truncation in both eventstream and remote runtime implementations

**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

OpenHands trajectory: https://www.all-hands.dev/share?share_id=904bcc7091f72bba47fa9c04cbef4109cf6b63b6e2c1728dccc2a0f718496425

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:d6cbdb6-nikolaik   --name openhands-app-d6cbdb6   docker.all-hands.dev/all-hands-ai/openhands:d6cbdb6
```

### CODE REVIEW

File: openhands/events/serialization/event.py [hunk 1]
```diff
@@ -111,20 +111,8 @@ def event_to_memory(event: 'Event', max_message_chars: int) -> dict:
 
     if 'extras' in d:
         remove_fields(d['extras'], DELETE_FROM_MEMORY_EXTRAS)
-    if isinstance(event, Observation) and 'content' in d:
-        d['content'] = truncate_content(d['content'], max_message_chars)
+    # Content truncation is now handled at the runtime level in action_execution_server.py
```
Comment (xingyaoww) [inline openhands/events/serialization/event.py:114]:
```suggestion
```

File: openhands/controller/agent_controller.py [hunk 2]
```diff
@@ -255,10 +255,7 @@ async def _handle_observation(self, observation: Observation):
             observation (observation): The observation to handle.
         """
         observation_to_print = copy.deepcopy(observation)
-        if len(observation_to_print.content) > self.agent.llm.config.max_message_chars:
-            observation_to_print.content = truncate_content(
-                observation_to_print.content, self.agent.llm.config.max_message_chars
-            )
+        # Content truncation is now handled at the runtime level in action_execution_server.py
```
Comment (xingyaoww) [inline openhands/controller/agent_controller.py:258]:
```suggestion
```

**Comment by enyst** [comment]
An alternative here could be to move this method in the Observation class. Then the runtime or other components using Observations can just make use of Observations with truncated content.🤔

**Comment by mamoodi** [comment]
@xingyaoww just a gentle nudge for this in case you had forgotten. Just let me know if you will continue with this :)

**Comment by xingyaoww** [comment]
Let's close this for now since i wasn't get OpenHands to complete this easily

**Pull request closed without merging**